python/cudf/cudf/_lib/column.pyx

```diff
@@ -352,12 +356,19 @@ cdef class Column:
             col = self.base_children[0]
         else:
             col = self
+
         data_dtype = col.dtype
-        cdef libcudf_types.type_id tid = <libcudf_types.type_id> (
-            <underlying_type_t_type_id> (
-                np_to_cudf_types[np.dtype(data_dtype)]
+        cdef libcudf_types.type_id tid
+
+        if not is_list_dtype(self.dtype):
+            tid = <libcudf_types.type_id> (
+                <underlying_type_t_type_id> (
+                    np_to_cudf_types[np.dtype(data_dtype)]
```

Comment:
Should we move the `ListType` into this dictionary even though it's not a numpy type?

Reply:
@brandon-b-miller, any thoughts here?

Reply:
TL:DR, probably fine to leave this as-is pending dtype work. 

If all we were doing was just keying into a dict to map to the right libcudf type, I wouldn't see a reason not to just add it to the dict - I don't think there is a reason the keys have to be numpy dtype objects, and the dict could easily be named `cudf_python_to_libcudf_types`.

The problem really is the call to `np.dtype` which will throw if a `ListDtype` comes along, or anything else that isn't on [this list](https://numpy.org/doc/stable/reference/generated/numpy.dtype.html) for that matter. This is the same problem you get with a theoretical `cudf.StringDtype` or any pandas dtypes as well. Since we have this pattern all over the cython I feel it boils down to the following three options:

- Have a branch for this case every time
- Make a cython function that concisely takes a cuDF python dtype and produces a libcudf dtype and errors usefully
- Make `cudf.dtype` object instances inherit from a basic cython class that carries the `.libcudf_dtype` as an attribute you can access from cython and apply the logic to find out what that is on init

I personally think option 3 would be the most useful from a programming perspective, but in any case solving that problem is a broader issue than this PR and just keeping the `if not` is the simplest thing to do in lieu of a more global solution.